app/frontend/src/routes.ts

```diff
@@ -15,17 +15,39 @@ export const confirmChangeEmailRoute = "/confirm-email";
 
 export const signupRoute = "/signup";
 
-export const userRoute = "/user";
-export const aboutRoute = `${userRoute}/about`;
-export const homeRoute = `${userRoute}/home`;
-export const referencesRoute = `${userRoute}/references`;
-export const favoritesRoute = `${userRoute}/favorites`;
-export const photosRoute = `${userRoute}/photos`;
-
-export const editProfileRoute = `${userRoute}/edit`;
-export const editHostingPreferenceRoute = `${userRoute}/edit-hosting`;
-
-export const routeToUser = (username: string) => `${userRoute}/${username}`;
+// user
+
+const userBaseRoute = "/user";
+export type UserTab = "about" | "home" | "references" | "favorites" | "photos";
+export type EditUserTab = Extract<UserTab, "about" | "home">;
+
+export const userRoute = `${userBaseRoute}/:userId?/:tab?`;
+export const editUserRoute = `${userBaseRoute}/edit/:tab?`;
+
+type UserRouteOptions<T extends UserTab> = {
+  username?: string;
+  tab?: T;
+  edit?: T extends EditUserTab ? true : T extends undefined ? true : never;
+};
+
+export function routeToUser<T extends EditUserTab>(
+  options?: Omit<UserRouteOptions<T>, "username"> & { edit: true }
+): string;
+export function routeToUser<T extends UserTab>(
+  options?: Omit<UserRouteOptions<T>, "edit">
+): string;
+export function routeToUser<T extends UserTab>(options?: T): string;
+export function routeToUser<T extends UserTab>({
+  username,
+  tab,
+  edit,
+}: UserRouteOptions<T> = {}) {
+  // edit: user/edit(/tab)
+  // view: user(/username)(/tab)
+  return `${userBaseRoute}${!edit && username ? "/" : ""}${
+    (!edit && username) || ""
+  }${edit ? "/edit" : ""}${tab ? "/" : ""}${tab || ""}`;
```

Comment:
```suggestion
  return `${userBaseRoute}${edit ? "/edit" : username ? `/${username}`: ""}${tab ? `/${tab}` : ""}`;
```
I think this would also work? And reduce the need of having to do the same boolean check twice